sui-execution/latest/sui-adapter/src/adapter.rs

```diff
@@ -83,7 +86,19 @@ mod checked {
     pub fn new_move_vm(
         natives: NativeFunctionTable,
         protocol_config: &ProtocolConfig,
+        _enable_profiler: Option<PathBuf>,
     ) -> Result<MoveVM, SuiError> {
+        #[cfg(feature = "gas-profiler")]
+        let vm_profiler_config = _enable_profiler.map(|_| VMProfilerConfig {
+            full_path: _enable_profiler.filter(|p| {
```

Comment:
Adding to my earlier comment about pushing the default output path logic up -- we should push it up even higher than this, so that if the caller to `new_move_vm` wants to start the profiler, they need to supply a valid path.